lib/quill_delta.dart

```diff
@@ -30,65 +32,69 @@ class Operation {
       _attributes == null ? null : new Map<String, dynamic>.from(_attributes);
   final Map<String, dynamic> _attributes;
 
-  Operation._(this.key, this.length, this.data, Map attributes)
-      : assert(key != null && length != null && data != null),
+  Operation._(this.operationType, this.length, this.data, Map attributes)
+      : assert(operationType != null && length != null && data != null),
         assert(() {
-          if (key != 'insert') return true;
+          if (operationType != OperationType.insert) return true;
           return data.length == length;
         }(), 'Length of insert operation must be equal to the text length.'),
         _attributes = attributes != null
             ? new Map<String, dynamic>.from(attributes)
             : null;
 
+  static String getOperationTypeString(OperationType opType) {
+    return opType.toString().split('.').last;
+  }
```

Comment:
This is fragile and the main reason I decided not to use Dart's enum in this case (I had it implemented as enum originally). What if future version of Dart changes `toString` representation of enums?

For now, I feel like enums are not the best choice when it comes to representing serializable types as they require this kind of (annoying) boilerplate.

Reply:
My preference would be to just simply declare constants and replace all hardcoded strings with those constants. E.g.:

```dart
class Operation {
  /// Key of insert operations.
  static const String insertKey = 'insert';

  /// Key of delete operations.
  static const String deleteKey = 'delete';

  /// Key of retain operations.
  static const String retainKey = 'retain';

  // ... rest
}
```

I feel like extra class for operation types provides little value comparing to number of lines of code added.

Would this work for you?

Reply:
obviously better than now, but not perfect )
because key is string and it not validate values

Reply:
The only way to create an instance of `Operation` is by using public factory constructors `Operation.insert`, `Operation.delete` and `Operation.retain` or with `Opeartion.fromJson`

All of this methods validate their inputs so there is no chance for someone to create an instance of Operation with invalid key.

Even when using an enum or a custom class the same validation would have to be implemented just in a different place. So in this case I prefer something that's simpler and less verbose.